pkg/pgmodel/metrics/database/database.go

```diff
@@ -0,0 +1,180 @@
+package database
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/pgxconn"
+)
+
+type Engine interface {
+	Run()
+	IsRunning() bool
+}
+
+type metricsEngineImpl struct {
+	conn      pgxconn.PgxConn
+	ctx       context.Context
+	isRunning atomic.Value
+	metrics   []metricQueryWrap
+}
+
+// NewEngine creates an engine that performs database metrics evaluation every evalInterval.
+// The engine runs predefined queries that returns a BIGINT as the metric value
+// which is then set as a value to Prometheus metric.
+//
+// Note: Make sure to call this only when the database is TimescaleDB. Plain Postgres
+// will cause evaluation errors.
+func NewEngine(ctx context.Context, conn pgxconn.PgxConn) *metricsEngineImpl {
+	engine := &metricsEngineImpl{
+		conn:    conn,
+		ctx:     ctx,
+		metrics: metrics,
+	}
+	engine.isRunning.Store(false)
+	return engine
+}
+
+func (e *metricsEngineImpl) register() {
+	prometheus.MustRegister(getMetrics(e.metrics)...)
+}
+
+func (e *metricsEngineImpl) unregister() {
+	m := getMetrics(e.metrics)
+	for i := range m {
+		prometheus.Unregister(m[i])
+	}
+}
+
+func getMetrics(m []metricQueryWrap) []prometheus.Collector {
+	var metrics []prometheus.Collector
+	for i := range m {
+		metrics = append(metrics, m[i].metric)
+	}
+	return metrics
+}
+
+const (
+	timeout       = time.Minute
+	evalInterval  = time.Minute * 3
+	maxRetryWait  = time.Minute * 30
+	stopThreshold = 5
+)
+
+func (e *metricsEngineImpl) Run() {
+	e.register()
```

Comment:
Let's check if engine is running using `isRunning`. The idea is to not allow starting engine twice